### PR TITLE
Add clang-format fixer for C/C++

### PR DESCRIPTION
**Pull request by renstrom**
All the paths, variables and functions use "c" instead of "cpp". Not sure what's preferred, or if I should duplicate it?

https://clang.llvm.org/docs/ClangFormat.html

### CODE REVIEW

**Comment by renstrom** [comment]
Added documentation now (I knew I missed something!). Not entirely sure I did it correct though.

File: doc/ale-cpp.txt [hunk 1]
```diff
@@ -190,4 +190,23 @@ g:ale_cpp_gcc_options                                   *g:ale_cpp_gcc_options*
 
 
 ===============================================================================
+clang-format                                              *ale-cpp-clangformat*
+
+g:ale_c_clangformat_executable                 *g:ale_c_clangformat_executable*
```
Comment (w0rp) [inline doc/ale-cpp.txt]: I think here you can just write `See |ale-c-clangformat|` and a short message about the options being the same.

Comment (renstrom) [inline doc/ale-cpp.txt]: Does this sound okay?

```
See |ale-c-clangformat| for information about the available options.
Note that the C options are also used for C++.
```

Comment (w0rp) [inline doc/ale-cpp.txt]: Yeah, that sounds fine to me.

**Comment by renstrom** [comment]
Done and done!

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**